include/alpaka/atomic/AtomicUniformCudaHip.hpp

```diff
@@ -0,0 +1,474 @@
+/* Copyright 2022 René Widera
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) || defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+
+#    include <alpaka/atomic/Op.hpp>
+#    include <alpaka/core/BoostPredef.hpp>
+#    include <alpaka/core/Positioning.hpp>
+
+#    include <type_traits>
+
+namespace alpaka
+{
+    //! The GPU CUDA/HIP accelerator atomic ops.
+    //
+    //  Atomics can be used in the hierarchy level grids, blocks and threads.
+    //  Atomics are not guaranteed to be save between devices.
+    class AtomicUniformCudaHipBuiltIn
+    {
+    };
+} // namespace alpaka
+
+#    if !defined(ALPAKA_HOST_ONLY)
+
+#        if defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !BOOST_LANG_CUDA
+#            error If ALPAKA_ACC_GPU_CUDA_ENABLED is set, the compiler has to support CUDA!
+#        endif
+
+#        if defined(ALPAKA_ACC_GPU_HIP_ENABLED) && !BOOST_LANG_HIP
+#            error If ALPAKA_ACC_GPU_HIP_ENABLED is set, the compiler has to support HIP!
+#        endif
+
+//! Inline namespace is required to check the existenz of function in global namespace via SFINAE.
+inline namespace alpakaGlobal
+{
+    //! Provide an interface to builtin atomic functions.
+    //
+    // To check for the existence of builtin functions located in the global namespace :: directly.
+    // This is forbidden by C++.
```

Comment:
I do not understand this line. Did you mean smth like
```suggestion
    // This would not be possible without having these types in global namespace.
```
?

Reply:
Yes! Your suggestion is more clear!